### DISCUSSION TITLE
Extract function call result from automatic function calling

### DISCUSSION THREAD

**Discussion by HuskyDanny** [microsoft/semantic-kernel#9602] [General]
Hi team,

I am using function calling for optional RAG. I want to extract the result of executing the RAG plugin, but I could only see the function call message and its output in the chatHistory, which then is packed into the metadata of Function.result
![image](https://github.com/user-attachments/assets/ff172c47-f300-4d92-ac05-297eed42355a)

I could use the chatHistory to match the function result with the function call id manually, but want to ask if there is any alternatives here? I am using semantic-kernel==1.7.0 in python.

**Comment by moonbox3** [comment]
Hi @HuskyDanny, thanks for your question. You'll want to look a bit deeper into the `ChatMessageContent` types -- the `FunctionCallContent` or `FunctionResultContent` will be included in the `items` collection. Take a look at the following discussion, please: #6589, specifically this [part](https://github.com/microsoft/semantic-kernel/discussions/6589#discussioncomment-9753757).
